hpcbench/driver.py

```diff
@@ -542,9 +544,26 @@ def _extract_metrics(self, **kwargs):
                 MetricsDriver(self.campaign, self.benchmark)(**kwargs)
         self.gather_metrics(runs)
 
+    def _add_build_info(self, execution):
+        executable = execution['command'][0]
+        try:
+            exepath = find_executable(executable, required=True)
+            if magic.from_file(exepath).startswith('ELF'):
+                if 'metas' not in execution or execution['metas'] is None:
+                    execution['metas'] = dict()
+                execution['metas']['build_info'] = extract_build_info(exepath)
+            else:
+                LOGGER.info('{xp} is not pointing to an ELF executable'.format(
+                            xp=exepath))
+        except NameError as ne:
```

Comment:
The `except` directive should cover as least code as possible:
```python
try:
  execpath = find_executable(...)
except NameError as ne:
  LOGGING.info('%s is not point to an ELF executable', exepath)
else:
  if magic.from_file(exepath) ..:
  ...
```

Note the `LOGGER.info` change that uses parameter expansion provided by `logging` module instead of building the message before logging it.